app/jobs/risc_delivery_job.rb

```diff
@@ -53,8 +53,6 @@ def perform(
       user:,
     )
   rescue *NETWORK_ERRORS => err
-    raise err if self.executions < 2 && !inline?
```

Comment:
Does removing this affect the retry logic? I think it may depend on the job failing.

Reply:
I was wondering about that as well. 

I also have a question [here](https://gitlab.login.gov/lg-people/lg-people-appdev/Melba/backlog-fy24/-/issues/37#note_175212) about whether we would also like to reduce the events logged as well.

The best approach seems to be:
Provide a block to `retry_on` to log the event after the last attempt fails. Don't do any `rescue` within the `perform`.

Reply:
These 2 statements in the Rails ActiveJob guide do seem to indicate that @mitchellhenke is right:
> If an exception from a job is not rescued, then the job is referred to as "failed".

> A failed job will not be retried, unless configured otherwise.

Reply:
Logging is fine to keep for every try imo

Reply:
Looking at this, and looking at the [ActiveJob source code for `retry_on`](https://github.com/rails/rails/blob/2ebb508cd8ee16c0bb280f91b93c01d939fcbf12/activejob/lib/active_job/exceptions.rb#L62), I think the way to stop reporting exceptions to NewRelic would be to add a block to our `retry_on` up top and make it a no-op, because the block is only called on the last/final time, so providing a block would be tell it to not keep raising

Reply:
From the `GoodJob` documentation:
> When using retry_on with a limited number of retries, the final exception will not be rescued and will raise to GoodJob's error handler. To avoid this, pass a block to retry_on to handle the final exception instead of raising it to GoodJob:
```
class ApplicationJob < ActiveJob::Base
  retry_on StandardError, attempts: 5 do |_job, _exception|
    # Log error, do nothing, etc.
  end
  # ...
end
```

Perhaps the right thing to do is to leave the existing `rescue` handler in, and use the block argument to `retry_on` to do nothing. This would work if the errors we are seeing on NewRelic is only _after the retries have failed_. I'll verify that this is true.

Reply:
Jinx